bucket_registry.go

```diff
@@ -35,17 +35,26 @@ func init() {
 	}
 }
 
-// registryBucket adds a newly opened Bucket to the registry.
-func (r *bucketRegistry) registerBucket(bucket *Bucket) {
+// registryBucket adds a newly opened Bucket to the registry. Returns true if the bucket already exists, and a copy of the bucket to use.
+func (r *bucketRegistry) registerBucket(bucket *Bucket) (bool, *Bucket) {
 	name := bucket.GetName()
 	debug("registerBucket %v %s at %s", bucket, name, bucket.url)
```

Comment:
This logging is done in _registerBucket, looks like you could remove it here.

Reply:
done.